integration_tests/conftest.py

```diff
@@ -265,4 +342,68 @@ def fuzzy_checker() -> FuzzyChecker:
 
     yield checker
 
-    checker.write_output()
+    checker.save_diagnostic_output()
+
+
+# Utilities for working with "multiplicative drifts" -- these are
+# pretty specific to this simulation, where we have a lot of fuzzy checks
+# that start out precise at the first timestep and gradually become
+# fuzzier as the simulation progresses.
+# We express these as multiplicative factors applied to the target value
+# per unit time.
+
+
+def multiplicative_drift_to_bound_at_timestep(
+    value: float, drift_per_timestep: float, num_timesteps: int
+) -> float:
+    return value * pow(drift_per_timestep, num_timesteps)
+
+
+def multiplicative_drift_to_bounds_at_timestep(
```

Comment:
Nit: shouldn't this be `drifts` not `bounds`? i.e.
```suggestion
def multiplicative_drifts_to_bound_at_timestep(
```

Reply:
There are two drift values input, and two bound values output. So maybe it's both -- `multiplicative_drifts_to_bounds_at_timestep`.